include/ast.hpp

```diff
@@ -108,6 +108,21 @@ struct IfStmtNode : public StmtNode {
   std::unique_ptr<StmtNode> or_else;
 };
 
+struct WhileStmtNode : public StmtNode {
+  WhileStmtNode(std::unique_ptr<ExprNode> predicate,
+                std::unique_ptr<StmtNode> loop, bool do_while = false)
```

Comment:
```suggestion
                std::unique_ptr<StmtNode> body, bool is_do_while = false)
```
- It would be more straightforward to say "the body of a while loop" instead of "the loop body of the while loop". :grin:
- For variables and functions that have a type or return a type `bool`, we typically begin the name with "is" or "has". This improves readability. Consider the following comparision:
    ```cpp
    // Read: "if the while statement is do-while"
    if (while_stmt.is_do_while) {}

    // Read: "if the while statement do-while"
    // Is this call on do_while "do"ing anything?
    if (while_stmt.do_while) {}
    ```